### ISSUE TITLE
Unable to switch users?

### ISSUE THREAD

**Issue by helloxz** [kingToolbox/WindTerm#329]
When I tried to change the user in the session, but it didn't work, it was still the last user.

* OS:windows 10
* windterm version:2.2

![](https://i.bmp.ovh/imgs/2021/12/328423301beaa914.png)

**Comment by kingToolbox** [comment]
Thank you, but I just tested and confirmed that there should be no problem. Can you try this again? Be careful not to use `Reconnect Session` button or `Return` key to reconnect the session after modification. Instead, you need to open a new session or close the session and then reopen it. `Reconnect Session` and `Return` key will only use the old configuration data.

**Comment by kingToolbox** [comment]
Thank you, but I just tested and confirmed that there should be no problem. Can you try this again? Be careful not to use `Reconnect Session` button or `Return` key to reconnect the session after modification. Instead, you need to open a new session or close the session and then reopen it. `Reconnect Session` and `Return` key will only use the old session configuration.

**Comment by helloxz** [comment]
Hello, I changed the user name and restarted windterm, but it still doesn't work.

1. Change the root user to the test user
![](https://i.bmp.ovh/imgs/2021/12/7b18017be4f65ef0.png)
2. save setting and restart windterm
3. Connect session again
4. Still use the root user instead of the test user
![](https://i.bmp.ovh/imgs/2021/12/8492c55270ab9145.png)

Note: I upgraded from version 2.1 to version 2.2 step by step

**Comment by helloxz** [comment]
Have you reproduced this BUG yet?

**Comment by kingToolbox** [comment]
Yes, I am fixing this issue.

By the way, did you use automatic login? After you clear the saved automatic login information (Session Settings Dialog - SSH - SSH - Authentication - Saved automatic authentication  - Clear), it should be fine.

**Comment by helloxz** [comment]
> Yes, I am fixing this issue.
> 
> By the way, did you use automatic login? After you clear the saved automatic login information (Session Settings Dialog - SSH - SSH - Authentication - Saved automatic authentication - Clear), it should be fine.

Yes, I used automatic login. Thank you for your proposal and look forward to the follow-up repair of this BUG

**Comment by kingToolbox** [comment]
This issue has been fixed. Now when you modify the `host`, the existing automatic login information will be cleared at the same time.

It is expected to be released in the new `WindTerm 2.3.0 Prerelease 3` version this weekend, please stay tuned, thank you.

**Comment by helloxz** [comment]
> This issue has been fixed. Now when you modify the `host`, the existing automatic login information will be cleared at the same time.
> 
> It is expected to be released in the new `WindTerm 2.3.0 Prerelease 3` version this weekend, please stay tuned, thank you.

Thank you for your hard work.

**Comment by kingToolbox** [comment]
I am happy to inform you that [WindTerm_2.3.0_Prerelease_3](https://github.com/kingToolbox/WindTerm/releases/tag/2.3-prerelease) has been released and this issue has been fixed, please download and check it, thank you.

**Comment by kingToolbox** [comment]
SORRY! `WindTerm_2.3.0_Prerelease_2` and `WindTerm_2.3.0_Prerelease_3` has a **serious SFTP bug** #487 and may cause file corruption after **uploaded**.

Please use the old versions or update to the [WindTerm_2.3.0_Prerelease_4](https://github.com/kingToolbox/WindTerm/releases/tag/2.3-prerelease) version. Thank you.

**Comment by winlong** [comment]
我在想怎么不能象SecureCRT一样可以在SSH里设置用户呢，想换一个用户登陆，一直找不到地方，原来是要先清除才行。
